runtime/compiler/il/symbol/J9Symbol.cpp

```diff
@@ -271,6 +271,44 @@ J9::Symbol::getRecognizedField()
       return TR::Symbol::UnknownField;
    }
 
+/**
+ * Return the owning class name of this recognized field.
+ * Return null if this symbol does not have a recognized field.
+ */
+const char *
+J9::Symbol::owningClassNameCharsForRecognizedField(int32_t & length)
+   {
+   length = 0;
+   TR::Symbol::RecognizedField recognizedField = getRecognizedField();
+   if (TR::Symbol::UnknownField == recognizedField)
+      {
+      return NULL;
+      }
+   int i;
```

Comment:
why is `i` not scoped to the loops? I don't see that it should carry across the loops...

Reply:
I tried to initialize both variables in the loop condition, but compiler complained. This is OK if both variables are of same type, but there seems to be a limitation if they are different types.

Reply:
Also declare just one of them inside the loop doesn't work either.
`for (int i =0, knownField = &recognizedFieldName_c[i]; ...`
As compiler thinks `knownField` is also an int.

Reply:
There are a few alternative choices - probably the best is to do something like:
```
for (int i = 0; recognizedFieldName_c[i].id != TR::Symbol::UnknownField; ++i) 
     {
     if (recognizedFieldName_c[i].id == recognizedField) {
        {
        ...
        return ...
        }
     }
```
The variables crossing scopes is bad form for maintenance. You could create the knownField temp within the loop body to simplify the code there and that would only be one load and check in the loop header and an additional load in the loop which is worth it for the maintenance benefits IMO.

Reply:
Simplified the loop conditions.